chat/client-api.md

```diff
@@ -0,0 +1,52 @@
+# Client API
+
+```typescript
+abstract class Client {
+  // ---------- Methods ----------------------------------------------- //
+
+  // initializes the client with persisted storage and a network connection
+  public abstract init(): Promise<void>;
+
+  // register a blockchain account with a public key and returns the public key
+  public abstract register(params: {
+    account: string;
+  }): Promise<string>;
+
+  // queries the default registry with a blockchain account and returns the public key
+  public abstract resolve(params: {
+    account: string;
+  }): Promise<string>;
+
+  // sends a chat invite with opening message and returns an invite id
+  public abstract invite(params: {
+    publicKey: string;
+    openingMessage: string;
+  }): Promise<number>;
```

Comment:
What would be an `invite id` use case?

Reply:
Perhaps tracking pending invites on the UI